src/plugins/visualizations/public/utils/saved_visualize_utils.ts

```diff
@@ -0,0 +1,392 @@
+
+      if (config) {
+        return {
+          ...config.toListItem(savedObject),
+          references: savedObject.references,
+        };
+      } else {
+        return mapHitSource(visTypes, savedObject);
+      }
+    }),
+  };
+}
+
+export async function getSavedVisualization(
+  services: {
+    savedObjectsClient: SavedObjectsClientContract;
+    search: DataPublicPluginStart['search'];
+    dataViews: DataPublicPluginStart['dataViews'];
+    spaces: SpacesPluginStart;
+    savedObjectsTagging?: SavedObjectsTaggingApi;
+  },
+  opts?: any
+): Promise<VisSavedObject> {
+  if (typeof opts !== 'object') {
+    opts = { id: opts };
+  }
+
+  const id = (opts.id as string) || '';
+  const savedObject = {
+    id,
+    copyOnSave: false,
+    migrationVersion: opts.migrationVersion,
+    displayName: SAVED_VIS_TYPE,
+    getEsType: () => SAVED_VIS_TYPE,
+    getDisplayName: () => SAVED_VIS_TYPE,
+    searchSource: opts.searchSource ? services.search.searchSource.createEmpty() : undefined,
+  } as VisSavedObject & { _source: any };
+  const config = { injectReferences };
+
+  const defaultsProps = getDefaults(opts);
+
+  if (!id) {
+    Object.assign(savedObject, defaultsProps);
+    return savedObject;
+  }
+
+  const {
+    saved_object: resp,
+    outcome,
+    alias_target_id: aliasTargetId,
+  } = await services.savedObjectsClient.resolve<SavedObjectAttributes>(SAVED_VIS_TYPE, id);
+
+  if (!resp._version) {
+    throw new SavedObjectNotFound(SAVED_VIS_TYPE, id || '');
+  }
+
+  savedObject._source = _.cloneDeep(resp.attributes);
+
+  if (savedObject._source.visState) {
+    savedObject._source.visState = JSON.parse(savedObject._source.visState);
+  }
+
+  // assign the defaults to the response
+  _.defaults(savedObject._source, defaultsProps);
+
+  Object.assign(savedObject, savedObject._source);
+  savedObject.lastSavedTitle = savedObject.title;
+
+  savedObject.sharingSavedObjectProps = {
+    aliasTargetId,
+    outcome,
+    errorJSON:
+      outcome === 'conflict' && services.spaces
+        ? JSON.stringify({
+            targetType: SAVED_VIS_TYPE,
+            sourceId: id,
+            targetSpace: (await services.spaces.getActiveSpace()).id,
+          })
+        : undefined,
+  };
+
+  const meta = savedObject._source.kibanaSavedObjectMeta || {};
+
+  if (meta.searchSourceJSON) {
+    try {
+      let searchSourceValues = parseSearchSourceJSON(meta.searchSourceJSON);
+
+      if (opts.searchSource) {
+        searchSourceValues = injectSearchSourceReferences(
+          searchSourceValues as any,
+          resp.references
+        );
+        savedObject.searchSource = await services.search.searchSource.create(searchSourceValues);
+      } else {
+        savedObject.searchSourceFields = searchSourceValues;
+      }
+    } catch (error: any) {
+      throw error;
+    }
+  }
+
+  if (resp.references && resp.references.length > 0) {
+    config.injectReferences(savedObject, resp.references);
```

Comment:
`config` is just an object containing `injectReferences`? Let's use it directly.

Reply:
done